src/main/kotlin/org/sol4k/TransactionMessage.kt

```diff
@@ -0,0 +1,368 @@
+package org.sol4k
+
+import org.sol4k.Constants.PUBLIC_KEY_LENGTH
+import org.sol4k.instruction.CompiledInstruction
+import org.sol4k.instruction.Instruction
+import java.io.ByteArrayOutputStream
+
+data class TransactionMessage(
```

Comment:
Let's make all the properties of `TransactionMessage` and it's constructor as private (needs conversion from data class to a normal class). The reason behind it is that the end user doesn't need to have access to the internals of a message. What they really need is only to call `MessageVersion.newMessage()`, `serialize()` and `deserialize()`. This ways the users would not be confused with the details of what has to be passed to the constructor and we will be able to change it's internals without breaking the clients code.